Core: Fix `--static-dir` with absolute path on Windows

ghengeveld opened the pull request:
Issue: #13336

## What I did

Primarily this fixes the handling of the `-s` (`--static-dir`) flag for Windows users. This flag accepts a string in the shape of `<dirname>:<endpoint>`. Currently it fails to handle absolute paths on Windows: for a value like `C:\path` it would assume `C` to be the dirname and `\path` the endpoint. That bug is addressed here.

Besides fixing the bug at hand, I've extracted this piece of logic into a util, because it previously existed in two separate places. The util is now under unit test and the surrounding code has been cleaned up with some nicer logging. I've also replaced `shelljs` with `copy` from `fs-extra`. One less dependency 📉 

## How to test

- Is this testable with Jest or Chromatic screenshots? No
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this.

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

shilman committed:
Merge pull request #13081 from redian/patch-1

Update storybook-preview-head-example.html.mdx

jonniebigodes committed:
chore:updates the theming documentation

yannbf committed:
docs(MIGRATION): add preset information for addon-backgrounds in 6.1

thebuilder committed:
refactor: replace inquirer with prompts

thebuilder committed:
refactor: use the correct initial value type

The value is false, so could also argue it should just be removed.

ThibaudAV committed:
feat: add enableProdMode() in RenderNgAppService when NODE_ENV !== 'development'

thebuilder committed:
add @types/prompts

ThibaudAV committed:
feat: add function to find Inputs & Outputs in an Angular component

The test allows to verify the function by using angular engine

ThibaudAV committed:
refactor: rework story's props association with the component's properties

- allow story with component to override local properties
- handles Input with `bindingPropertyName`

ThibaudAV committed:
fix: angluar-cli export Chip component to be accessible by storybook render module

andyrichardson committed:
Fix missing `done` attribute on type definition

ghengeveld committed:
Extract static file handling to utils and make sure we handle Windows well.

ghengeveld committed:
We no longer need shelljs.

shilman reviewed (changes requested):
small thing: `.test.js` => `.test.ts`

ghengeveld committed:
Fix IE11 compatibility by avoiding ES6 syntax and using XHR instead of fetch.

ghengeveld committed:
err can be undefined.

ghengeveld committed:
Convert to TS.

ghengeveld committed:
Merge branch 'next' into 13336-windows-static-dir

ghengeveld commented:
> small thing: `.test.js` => `.test.ts`

Done

thebuilder committed:
fix types in initiate

thebuilder committed:
update lock file

kylegach committed:
UI: Fix display of sidebar logo

shilman committed:
sort packages

shilman committed:
MIGRATION: Note about react/react-dom peer dependencies

shilman committed:
Update MIGRATION.md

gaetanmaisse committed:
Merge pull request #13343 from brandon-leapyear/patch-1

Addon-storyshots: Add React as peer dependency

thebuilder committed:
remove unused groups

ThibaudAV committed:
test: add test for RenderNgAppService

Configure app/angular package as jest projects
Allow to use jest-preset-angular for app/angular tests

ThibaudAV committed:
feat: add a rewrite of client angular rendering

A `useLegacyRendering` flag allows to use old rendering engine

shilman committed:
MIGRATION.md for new angular renderer

shilman committed:
6.2.0-alpha.2 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.2.0-alpha.2

shilman committed:
v6.2.0-alpha.2

shilman committed:
Update git head to 6.2.0-alpha.2

shilman committed:
6.2.0-alpha.2 next.json version file

shilman committed:
Update MIGRATION.md

shilman committed:
Update MIGRATION.md

shilman committed:
Update MIGRATION.md

jonniebigodes committed:
added aside to mention the no manager flag.
Will unblock the user

HorusGoul committed:
CLI: fix generated stories path

The path where Storybook generates the boilerplate changed from `storybook-examples` to `stories`
in https://github.com/storybookjs/storybook/blob/next/lib/cli/src/helpers.ts#L209-L214, but wasn't
updated in the `Introduction.stories.mdx` (which also happens to correspond with the old
`welcome.mdx` file that doesn't exist anymore).

With this change, we can prevent people from getting confused by not finding the `storybook-examples`
folder.

ghengeveld committed:
Add more stories for custom branding

ghengeveld committed:
Reduce spacing to support a larger brandImage

ghengeveld committed:
Fix image misalignment when logo has no link

Additionally this refactors the Brand component for readability.

shilman committed:
6.1.10 changelog

shilman committed:
6.2.0-alpha.3 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.2.0-alpha.3

shilman committed:
v6.2.0-alpha.3

shilman committed:
Update git head to 6.2.0-alpha.3

dependabot-preview[bot] committed:
Bump @ember/test-helpers from 1.7.1 to 2.1.0

Bumps [@ember/test-helpers](https://github.com/emberjs/ember-test-helpers) from 1.7.1 to 2.1.0.
- [Release notes](https://github.com/emberjs/ember-test-helpers/releases)
- [Changelog](https://github.com/emberjs/ember-test-helpers/blob/master/CHANGELOG.md)
- [Commits](https://github.com/emberjs/ember-test-helpers/compare/v1.7.1...v2.1.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

shilman committed:
6.2.0-alpha.3 next.json version file

shilman committed:
Addon-docs: Handle svelte docgen failures gracefully

shilman committed:
Fix failing tests

shilman committed:
6.2.0-alpha.4 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.2.0-alpha.4

shilman committed:
v6.2.0-alpha.4

shilman committed:
Update git head to 6.2.0-alpha.4

shilman committed:
6.2.0-alpha.4 next.json version file

shilman committed:
Merge branch 'next' into 13336-windows-static-dir

shilman reviewed (approved):
LGTM - nice tests 💯
